doc/nrf/ug_fw_update.rst

```diff
@@ -0,0 +1,311 @@
+
+For a full list of the supported types, use the ``--help`` argument with the tool or any of its commands.
+
+.. _ug_fw_update_keys_python:
+
+Python
+------
+
+The |NCS| ships with an internal script used for interacting with private/public keys with the |NSIB|.
+This script is maintained by the |NCS| team and utilizes the ``ecdsa`` package available from PyPI.
+As such, it is only valid for use with ECDSA keys.
+
+.. code-block:: console
+
+   python3 nrf/scripts/bootloader/keygen.py --private -o priv.pem
+
+.. _ug_fw_update_development_keys:
+
+Development keys
+================
+
+When testing the bootloader chain, generating and using custom signing keys is optional.
+If you do not provide your own keys through Kconfig options, the build system will automatically create debug keys, depending on the bootloader(s) compiled into the application.
+
+.. caution::
+
+   Keys that are automatically used or generated by bootloaders for image signature validation are intended for *development and/or debug use only*.
+
+   Applications should *never* go into production when they are not protected by secure keys.
+   It is strongly encouraged to create and store these keys in a safe location, not only to protect the security of the application but also to ensure that the hardware can receive firmware updates throughout the project lifecycle.
+
+   Moreover, while default keys for MCUboot are tracked in its repository, and are therefore publicly visible, the development/debug keys autogenerated by |NSIB| will change *whenever the build directory is rebuilt*, that is, building without the build directory already existing.
+   If a build directory is removed and rebuilt from scratch, relying on the default ECDSA key for signing an application or upgradable second-stage bootloader image will result in a failed boot chain validation, if you are not programming also the |NSIB|.
+
+   You can avoid this issue by storing a custom private key outside of the build directory during development.
+
+.. _ug_fw_update_key_revocation:
+
+Public key revocation
+=====================
+
+The |NSIB| supports the revocation of public signing keys that are available to validate the next image in the secure boot chain.
+Key revocation can be an useful security or backup tool for devices that have already been deployed to the field.
+If a private key has been compromised or lost, its public key can be invalidated by uploading a new firmware image signed by another key known to the bootloader.
+
+These keys are kept internally by the bootloader, so the list of available public keys cannot change once it is deployed.
+See :option:`CONFIG_SB_PUBLIC_KEY_FILES` for details on how this mechanism is implemented.
+
+You can add this feature to your own project and check its functionality as follows:
+
+1. Use this :ref:`Python tool <ug_fw_update_keys_python>` to generate two or more private keys for the application and then extract a public key for each one:
```

Comment:
`this Python tool` does not make me think of the Python script described below. I was expecting a download link to a Python tool at the first moment. Maybe `the bootloader Python script`?